### PR TITLE
Add cross validation feature

### PR DESCRIPTION
**Pull request by JieXiong9119**

### CODE REVIEW

File: FDD_RF_Module/FDD_RF_Modeling.py [hunk 1]
```diff
@@ -86,10 +88,18 @@ def inputs_output_generator(self, train_or_test):
...
             for simulation_data_file_name in simulation_data_file_list:
                 temp_raw_FDD_data = pd.read_csv(f'data\\{self.weather}\\{self.weather}\\{simulation_data_file_name}')
+                #temp_raw_FDD_data = temp_raw_FDD_data.apply(pd.to_numeric, errors='coerce')
+                #temp_raw_FDD_data = temp_raw_FDD_data.dropna()
                 temp_raw_FDD_data = temp_raw_FDD_data.groupby(temp_raw_FDD_data.index // (self.aggregate_n_runs)).mean().iloc[:,0:-8]
                 temp_raw_FDD_data['label'] = meta_data.loc[meta_data.id == simulation_data_file_name[0:-12]].fault_type.values[0]
-                fault_inputs_output = pd.concat([fault_inputs_output, temp_raw_FDD_data], axis = 0)
-
+                # Splitting training and testing data
+                temp_raw_FDD_data_train, temp_raw_FDD_data_test = train_test_split(temp_raw_FDD_data, test_size=0.2, random_state=np.random.RandomState(self.randomseed))
```
Comment (JieXiong9119) [inline FDD_RF_Module/FDD_RF_Modeling.py:96]: `random_state=np.random.RandomState(self.randomseed)` make sure the reproducibility of random split.

File: FDD_RF_Module/FDD_RF_Modeling.py [hunk 2]
```diff
@@ -86,10 +88,18 @@ def inputs_output_generator(self, train_or_test):
...
+                fault_inputs_output = pd.concat([fault_inputs_output, temp_raw_FDD_data_train], axis = 0)
+                print('Split and save test data for ' + simulation_data_file_name)
+                temp_raw_FDD_data_test.to_csv(f'data\\testing_data\\{simulation_data_file_name}')
+
+            ind = pd.DataFrame(temp_raw_FDD_data_test.index.tolist())
+            ind.to_csv('ind.csv')
```
Comment (JieXiong9119) [inline FDD_RF_Module/FDD_RF_Modeling.py:102]: `ind.csv` records the indices selected for test dataset.

File: run/test_run.py [hunk 3]
```diff
@@ -5,7 +5,8 @@
     labeling_methodology = 'Simple',
     feature_selection_methodology = 'Embedded',
     aggregate_n_runs = 6 * 24,
-    number_of_trees = 2
+    number_of_trees = 4,
+    randomseed=20210813
```
Comment (JieXiong9119) [inline run/test_run.py:9]: Fixing `randomseed` will result in the same output.

File: FDD_RF_Module/FDD_RF_Modeling.py (hunk 1 above)
Comment (JanghyunJK) [reply FDD_RF_Module/FDD_RF_Modeling.py:96]: ahh did not know there is an easy method for splitting data in sklearn package!

**Comment by JanghyunJK** [comment]
looks all good to me. working on visualization now. merging.

**Pull request merged**